apps-api/backoffice/src/modules/election/services.ts

```diff
@@ -0,0 +1,148 @@
+import { mapRepositoryError } from '@pple-today/api-common/utils'
+import {
+  Election,
+  ElectionEligibleBallot,
+  ElectionEligibleVoter,
+  ElectionType,
+  EligibleVoterType,
+} from '@pple-today/database/prisma'
+import Elysia from 'elysia'
+import { ok } from 'neverthrow'
+
+import { ElectionStatus, ListElectionResponse } from './models'
+import { ElectionRepository, ElectionRepositoryPlugin } from './repostiory'
+
+export class ElectionService {
+  constructor(private readonly electionRepository: ElectionRepository) {}
+
+  private filterOnsiteElection(election: Election): boolean {
+    const now = new Date()
+
+    const isPublished = election.publishDate && now > election.publishDate
+    if (!isPublished) {
+      return false
+    }
+
+    const SECONDS_IN_A_DAY = 60 * 60 * 24
+    const { startResult, endResult, closeVoting } = election
+    const isResultAnnounced = startResult && endResult
+
+    if (isResultAnnounced && endResult) {
+      const isPastAnnouncePeriod = now > endResult
+      if (isPastAnnouncePeriod) {
+        return false
+      }
+    } else {
+      const daysSinceClose = (now.getTime() - closeVoting.getTime()) / (SECONDS_IN_A_DAY * 1000)
+      const hasExceededGracePeriod = daysSinceClose > 7
+
+      if (hasExceededGracePeriod) {
+        return false
+      }
+    }
+
+    return true
+  }
+
+  private filterOnlineElection(election: Election): boolean {
```

Comment:
Please change to `filterActiveOnlineElection` to specify that this method is used for filter only active election